### ISSUE TITLE
Update e2e tests and deploy strategy

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/fecfile-web-app#1925]
Business Reason

As a developer, I will be able to deploy to production even if end-to-end tests are failing so that I can quickly address time-sensitive production issues.

After this is deployed to develop:

*   [x] turn off nightly e2e-test job in circleci environment
*   [x] run e2e tests when app repo merges into develop branch
*   [x] run e2e tests when api repo merges into develop, release, or main branches
*   [x] proceed with deploy on PROD for both app and api repos when any required jobs including e2e-test fails

### QA Notes ###

 
 null

### DEV Notes ###

 
  
It might make the most sense to fail develop and release branches, but not production. Relevant issue: #1928

The CircleCI config.yml file needs to be updated here to not had a requires deploy dependency on e2e-tests:

![Image](https://github.com/fecgov/fecfile-web-app/assets/2971785/ae851bf7-bc62-45c9-84b2-a4966210ba60)

### Design ###

 
 null

 See full ticket and images here: [FECFILE-393](https://fecgov.atlassian.net/browse/FECFILE-393)

Pull Request: https://github.com/fecgov/fecfile-web-app/pull/2034

**Comment by mjtravers** [comment]
Blocked by #2009 
Placing ticket on hold.

**Comment by mjtravers** [comment]
During testing, it was found that for target deployments that do not require e2e tests to pass, that the deployment was no longer failing for unit tests, linting, and dependency check. Sending ticket back to In Progress.

See screenshot of CircleCI dashboard:


![Image](https://github.com/fecgov/fecfile-web-app/assets/2971785/bb923174-8bea-4689-af5d-0a661cfb9d1a)

**Comment by mjtravers** [comment]
#2009 no longer blocking this ticket. Removing "on hold" tag.

**Comment by mjtravers** [comment]
Here are the settings of the CircleCI "e2e tests nightly" trigger should we need to restore it at some point in the future:

Complete form:
![Image](https://github.com/fecgov/fecfile-web-app/assets/2971785/317ee6a8-2ab9-4958-a018-01b4fe7a86ef)



Bottom section of form:
![Image](https://github.com/fecgov/fecfile-web-app/assets/2971785/32c26c4e-9127-4bf5-bc3a-9374197c9d4f)

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

Here is screenshot showing deploy to DEV dependency on successful e2e tests:


![Image](https://github.com/fecgov/fecfile-web-app/assets/2971785/af1f8b7e-cfc5-4eb1-a9db-c51ae27ae44e)
